browser/model/jbosseap.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+let path = require('path');
+let fs = require('fs');
+var rimraf = require('rimraf');
+
+import JbosseapAutoInstallGenerator from './jbosseap-autoinstall';
+import InstallableItem from './installable-item';
+import Installer from './helpers/installer';
+import Logger from '../services/logger';
+import JdkInstall from './jdk-install';
+import Platform from '../services/platform';
+
+class JbosseapInstall extends InstallableItem {
+  constructor(installerDataSvc, targetFolderName, downloadUrl, fileName, jbosseapSha256) {
+    super(JbosseapInstall.KEY, downloadUrl, fileName, targetFolderName, installerDataSvc, true);
+
+    this.sha256 = jbosseapSha256;
+    this.installConfigFile = path.join(this.installerDataSvc.tempDir(), 'jbosseap-autoinstall.xml');
+    this.ConfigFile = path.join(this.installerDataSvc.tempDir(), 'jbosseap-autoinstall.xml.variables');
+    this.runtimeproperties = path.join(this.installerDataSvc.installRoot, '\\developer-studio\\studio', 'runtime_locations.properties');
+    this.addOption('install', this.version, '', true);
+  }
+
+  static get KEY() {
+    return 'jbosseap';
+  }
+
+  installAfterRequirements(progress, success, failure) {
+    progress.setStatus('Installing');
+    this.installGenerator = new JbosseapAutoInstallGenerator(this.installerDataSvc.jbosseapDir(), this.installerDataSvc.jdkDir(), this.version);
+    let installer = new Installer(JbosseapInstall.KEY, progress, success, failure);
+
+    if(fs.existsSync(this.installerDataSvc.jbosseapDir())) {
+      return Promise.resolve().then(()=>{
```

Comment:
Using a return statement here very much ends the method right here if the folder exists. And just removing it will create a race condition.

The two promise chains need to be merged if they are both supposed to run. And please use fs-extra or rimraf to remove the folder - these are already promisified.